ledger.go

```diff
@@ -1714,3 +1669,64 @@ func (l *Ledger) storeTransactions(stored *Block, evicted *Block) error {
 
 	return nil
 }
+
+// filterInvalidVotes takes a slice of (*finalizationVote)'s and filters away
+// ones that are invalid with respect to the current nodes state.
+func (l *Ledger) filterInvalidVotes(current *Block, votes []Vote) {
+ValidateVotes:
+	for _, vote := range votes {
+		vote := vote.(*finalizationVote)
+
+		// Ignore nil block proposals.
```

Comment:
not sure about those comments, if statements already doing pretty good job explaining whats going on in each of them

Reply:
Some of the if statements are a little confusing, so the comments are just to give some clarifications.

For example: `tx.Block >= current.Index + uint64(conf.GetPruningLimit())`.